### ISSUE TITLE
Milestone 2: Model fine tuning and training

### ISSUE THREAD

**Issue by jose-izarra** [IERoboticsClub/ChatbotWeb#8]
We will use the medical dataset to fine-tune a base LLM (e.g., GPT-3.5) for improved performance on healthcare-related questions. The goal is to train the model using domain-specific knowledge to avoid hallucinations and keep safe outputs.

## Select base model

Choose fine-tuning target (e.g., gpt-3.5-turbo, mistral-7b, etc.)

Confirm model supports fine-tuning and token limits

## Set up fine-tuning environment

Choose provider (OpenAI API, Hugging Face)

Install required CLI tools or SDKs

Authenticate and configure access

Here is a [guide](https://platform.openai.com/docs/guides/fine-tuning) on fine tuning a gpt model with a lot of examples

##  Launch fine-tuning job

Upload dataset to model provide. 

Configure training parameters (batch size, epochs, learning rate)

Example: 
```
from openai import OpenAI
client = OpenAI()

client.fine_tuning.jobs.create(
    training_file="file-abc123",
    model="gpt-4o-mini-2024-07-18"
)
```

## Evaluate model performance

Use a validation set to check:

* Accuracy on common medical queries

* Tendency to hallucinate or give unsafe advice

* Proper formatting and clarity

This part will be a bit annoying since you will have to test the model to check its accuracy on answering questions.
